tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1179,6 +1180,46 @@ class ResizeMethod(object):
   MITCHELLCUBIC = 'mitchellcubic'
 
 
+def _check_equal_input_output_dtypes(func):
+  """This decorator issue will issue a warning if the input dtype and output
+  dtype are different. Help to prevent silent dtype conversion from integer to
+  floating point format."""
+
+  def wrapper(*args, **kwargs):
+    try:
+      check_dtype = _check_equal_input_output_dtypes._fns_flagged[func.__name__]
+    except KeyError:
+      _check_equal_input_output_dtypes._fns_flagged[func.__name__] = True
+      check_dtype = True
```

Comment:
Can we change this to the simpler (and more efficient, at least after we made this into a utility function):

```
if input_dtype == output_dtype:
  return

if funcname in _CHECK_EQUAL_INPUT_OUTPUT_FNS_FLAGGED:
  return

# print warning
_CHECK_EQUAL_INPUT_OUTPUT_FNS_FLAGGED.add(funcname)


```